client/components/AdminProducts.js

```diff
@@ -0,0 +1,123 @@
+import React, { Component } from 'react';
+import { Link } from 'react-router-dom';
+import { connect } from 'react-redux';
+import { fetchProducts, fetchCategories, searchProducts, deleteProduct } from '../store';
+
+class AdminDash extends Component {
+  constructor() {
+    super();
+    this.state = {
+      searchTerm: '',
+    };
+  }
+
+  componentDidMount() {
+    // eslint-disable-next-line no-shadow
+    const { fetchCategories, fetchProducts } = this.props;
+    fetchCategories();
+    fetchProducts();
+  }
+
+  enterSearch = ({ target }) => {
+    this.setState({ searchTerm: target.value });
+  };
+
+  // clear user search
+  clearSearch = evt => {
+    evt.preventDefault();
+    // eslint-disable-next-line no-shadow
```

Comment:
If you find a linter rule that seems overly restrictive or just does not make sense and you disable it for several lines, then perhaps it would be a good idea to bring it up to the team to see whether or not we can remove that rule.